backend/src/utils/githubUtil.ts

```diff
@@ -274,6 +274,23 @@ const validateOpen = async (
     }
   });
 
+/** Determines the overall status of the submission. */
+export const getSubmissionStatus = (
+  submission: DevPortfolioSubmission
+): 'pending' | 'valid' | 'invalid' => {
+  // if text area populated, set to pending
+  if (submission.text) {
+    return 'pending';
+  }
+
+  // otherwise, valid if at least one valid in each category
+  const atLeastOneValid =
+    submission.openedPRs.some((pr) => pr.status === 'valid') &&
+    submission.reviewedPRs.some((pr) => pr.status === 'valid');
+
+  return atLeastOneValid ? 'valid' : 'invalid';
+};
```

Comment:
I think this logic is better off in the API layer

Reply:
Why? This function is used primarily in this utils file, and its only exported for use in the DAO layer to make older submission types in the db backwards compatible with the new type. Once each submission in the db has the new type, then I can remove the use of it outside of this file.

Reply:
That's fair